src/server/services/utils/urlParams.util.ts

```diff
@@ -3,3 +3,17 @@ export function removeUndefinedValueInQueryParameterList(queryList: Record<strin
 		if (queryList[key.toString()] === '' && !queryList[key.toString()]) delete queryList[key];
 	});
 }
+
+const ALLOWED_QUERY_PARAMS_TYPE_LIST = ['string', 'number', 'boolean'];
+
+export function transformObjectToQueryString<T extends object>(source: T): string {
+	const urlSearchParams = new URLSearchParams();
+
+	Object.entries(source).forEach(([key, value]) => {
+		if(ALLOWED_QUERY_PARAMS_TYPE_LIST.includes(typeof value)) {
+			urlSearchParams.append(key, value.toString());
+		}
+	});
+
+	return urlSearchParams.toString();
+}

```

Comment:
Tu peux inverser le filtre et l'instanciation des SearchParams pour ne pas append à chaque fois : 
```suggestion
export function transformObjectToQueryString<T extends object>(source: T): string {
    const filteredParams = Object.entries(source).filter(([key, value]) => (
        ALLOWED_QUERY_PARAMS_TYPE_LIST.includes(typeof value)
    ))
    const urlSearchParams = new URLSearchParams(filteredParams);

    return urlSearchParams.toString();
}
```
Du coup tu n'utilise plus la `key` donc tu peux te servir de `Object.values()` plutôt qu'`Object.entries()` : 
```suggestion
export function transformObjectToQueryString<T extends object>(source: T): string {
    const filteredParams = Object.values(source).filter((value) => (
        ALLOWED_QUERY_PARAMS_TYPE_LIST.includes(typeof value)
    ))
    const urlSearchParams = new URLSearchParams(filteredParams);

    return urlSearchParams.toString();
}
```